### PR TITLE
.coafile: Check yaml files and the templates

### PR DESCRIPTION
**Pull request by RazvanMitrica**
This entrypoint adds a .coafile for .yaml
and .jinja2 files.

Closes https://github.com/coala/coAST/issues/33

### CODE REVIEW

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/home.html.jinja2, line 16.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/home.html.jinja2
+++ b/tmp/tmplz2jz9lw/templates/home.html.jinja2
@@ -13,7 +13,7 @@
   <ul class="thin-row-medium" id="languages-list">
   {% for language in languages %}
     <li><a href="{% url 'languages', language %}">{{ m.language_name(language) }}</a>
-  {% endfor %}
+  {% endfor %}{# for language in languages #}
   </ul>
 </body>
 </html>
```

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/Language.html.jinja2, line 17.

Variable blocks should be spaced with `1` spaces on each side.

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/Language.html.jinja2
+++ b/tmp/tmplz2jz9lw/templates/Language.html.jinja2
@@ -14,7 +14,7 @@
     <tr>
       <th>Grammar File</th>
       <td>
-        <a href = {{language.grammar_file}}>
+        <a href = {{ language.grammar_file }}>
           {{ m.show_grammar(language.grammar_file) }}
         </a>  
       </td>
```

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/macros.jinja2, line 6.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/macros.jinja2
+++ b/tmp/tmplz2jz9lw/templates/macros.jinja2
@@ -3,7 +3,7 @@
 {{ language.full_name }}
 {% else %}
 {{ language.identifier }}
-{% endif %}
+{% endif %}{# if language.full_name #}
 {% endmacro %}
 
 {% macro show_grammar(grammar_file) %}
```

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/macros.jinja2, line 10.

Variable blocks should be spaced with `1` spaces on each side.

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/macros.jinja2
+++ b/tmp/tmplz2jz9lw/templates/macros.jinja2
@@ -7,7 +7,7 @@
 {% endmacro %}
 
 {% macro show_grammar(grammar_file) %}
-  {{grammar_file}}
+  {{ grammar_file }}
 {% endmacro %}
 
 {% macro show_list(list, separator=',') %}
```

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/macros.jinja2, line 15.

Variable blocks should be spaced with `1` spaces on each side.

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/macros.jinja2
+++ b/tmp/tmplz2jz9lw/templates/macros.jinja2
@@ -12,7 +12,7 @@
 
 {% macro show_list(list, separator=',') %}
   {% for item in list %}
-    {{ item.pk | escape }}{% if not loop.last %}{{separator}} {% endif %}
+    {{ item.pk | escape }}{% if not loop.last %}{{ separator }} {% endif %}
   {% endfor %}
 {% endmacro %}
 
```

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/macros.jinja2, line 16.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/macros.jinja2
+++ b/tmp/tmplz2jz9lw/templates/macros.jinja2
@@ -13,7 +13,7 @@
 {% macro show_list(list, separator=',') %}
   {% for item in list %}
     {{ item.pk | escape }}{% if not loop.last %}{{separator}} {% endif %}
-  {% endfor %}
+  {% endfor %}{# for item in list #}
 {% endmacro %}
 
 {% macro show_delimiter_list(delimiters, type) %}
```

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/macros.jinja2, line 28.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/macros.jinja2
+++ b/tmp/tmplz2jz9lw/templates/macros.jinja2
@@ -25,7 +25,7 @@
         <tt>{{ item.end | escape }}</tt>
       {% else %}
         (EOL)
-      {% endif %}
+      {% endif %}{# if item.end #}
     </li>
     {% endif %}
   {% endfor %}
```

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/macros.jinja2, line 30.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/macros.jinja2
+++ b/tmp/tmplz2jz9lw/templates/macros.jinja2
@@ -27,7 +27,7 @@
         (EOL)
       {% endif %}
     </li>
-    {% endif %}
+    {% endif %}{# if item.type == type #}
   {% endfor %}
 </ul>
 {% endmacro %}
```

**Comment by jayvdb** [comment]
Comment on da8c4cbbe88bec566a958e5fa2327e142bbcfef4, file templates/macros.jinja2, line 31.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmplz2jz9lw/templates/macros.jinja2
+++ b/tmp/tmplz2jz9lw/templates/macros.jinja2
@@ -28,6 +28,6 @@
       {% endif %}
     </li>
     {% endif %}
-  {% endfor %}
+  {% endfor %}{# for item in delimiters #}
 </ul>
 {% endmacro %}
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/Language.html.jinja2, line 17.

Variable blocks should be spaced with `1` spaces on each side.

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/Language.html.jinja2
+++ b/tmp/tmpn_oayuko/templates/Language.html.jinja2
@@ -14,7 +14,7 @@
     <tr>
       <th>Grammar File</th>
       <td>
-        <a href = {{language.grammar_file}}>
+        <a href = {{ language.grammar_file }}>
           {{ m.show_grammar(language.grammar_file) }}
         </a>  
       </td>
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/macros.jinja2, line 6.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/macros.jinja2
+++ b/tmp/tmpn_oayuko/templates/macros.jinja2
@@ -3,7 +3,7 @@
 {{ language.full_name }}
 {% else %}
 {{ language.identifier }}
-{% endif %}
+{% endif %}{# if language.full_name #}
 {% endmacro %}
 
 {% macro show_grammar(grammar_file) %}
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/macros.jinja2, line 10.

Variable blocks should be spaced with `1` spaces on each side.

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/macros.jinja2
+++ b/tmp/tmpn_oayuko/templates/macros.jinja2
@@ -7,7 +7,7 @@
 {% endmacro %}
 
 {% macro show_grammar(grammar_file) %}
-  {{grammar_file}}
+  {{ grammar_file }}
 {% endmacro %}
 
 {% macro show_list(list, separator=',') %}
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/macros.jinja2, line 15.

Variable blocks should be spaced with `1` spaces on each side.

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/macros.jinja2
+++ b/tmp/tmpn_oayuko/templates/macros.jinja2
@@ -12,7 +12,7 @@
 
 {% macro show_list(list, separator=',') %}
   {% for item in list %}
-    {{ item.pk | escape }}{% if not loop.last %}{{separator}} {% endif %}
+    {{ item.pk | escape }}{% if not loop.last %}{{ separator }} {% endif %}
   {% endfor %}
 {% endmacro %}
 
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/macros.jinja2, line 16.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/macros.jinja2
+++ b/tmp/tmpn_oayuko/templates/macros.jinja2
@@ -13,7 +13,7 @@
 {% macro show_list(list, separator=',') %}
   {% for item in list %}
     {{ item.pk | escape }}{% if not loop.last %}{{separator}} {% endif %}
-  {% endfor %}
+  {% endfor %}{# for item in list #}
 {% endmacro %}
 
 {% macro show_delimiter_list(delimiters, type) %}
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/macros.jinja2, line 28.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/macros.jinja2
+++ b/tmp/tmpn_oayuko/templates/macros.jinja2
@@ -25,7 +25,7 @@
         <tt>{{ item.end | escape }}</tt>
       {% else %}
         (EOL)
-      {% endif %}
+      {% endif %}{# if item.end #}
     </li>
     {% endif %}
   {% endfor %}
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/macros.jinja2, line 30.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/macros.jinja2
+++ b/tmp/tmpn_oayuko/templates/macros.jinja2
@@ -27,7 +27,7 @@
         (EOL)
       {% endif %}
     </li>
-    {% endif %}
+    {% endif %}{# if item.type == type #}
   {% endfor %}
 </ul>
 {% endmacro %}
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/macros.jinja2, line 31.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/macros.jinja2
+++ b/tmp/tmpn_oayuko/templates/macros.jinja2
@@ -28,6 +28,6 @@
       {% endif %}
     </li>
     {% endif %}
-  {% endfor %}
+  {% endfor %}{# for item in delimiters #}
 </ul>
 {% endmacro %}
```

**Comment by jayvdb** [comment]
Comment on 3eb631373e7791cdd550ef869e187859f6bd628d, file templates/home.html.jinja2, line 16.

Unlabeled control end tag

*Origin: Jinja2Bear, Section: `jinja2`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn_oayuko/templates/home.html.jinja2
+++ b/tmp/tmpn_oayuko/templates/home.html.jinja2
@@ -13,7 +13,7 @@
   <ul class="thin-row-medium" id="languages-list">
   {% for language in languages %}
     <li><a href="{% url 'languages', language %}">{{ m.language_name(language) }}</a>
-  {% endfor %}
+  {% endfor %}{# for language in languages #}
   </ul>
 </body>
 </html>
```

File: views/languages.yaml [hunk 1]
```diff
@@ -5,6 +5,9 @@ path:
 template: Language
 context:
   dynamic:
-    delimiter_type_comment: session.query(EnumDelimiterType).filter(EnumDelimiterType.name == 'comment').one()
-    delimiter_type_string: session.query(EnumDelimiterType).filter(EnumDelimiterType.name == 'string').one()
-    delimiter_type_block: session.query(EnumDelimiterType).filter(EnumDelimiterType.name == 'block').one()
+    delimiter_type_comment: session.query(EnumDelimiterType).filter
+(EnumDelimiterType.name == 'comment').one()
+    delimiter_type_string: session.query(EnumDelimiterType).filter
```
Comment (jayvdb) [inline views/languages.yaml]: syntax error: could not find expected ':'

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/TinyBasic.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 identifier: TinyBasic
 full_name: TinyBasic
 wikidata: Q586622
-grammar_file: https://github.com/antlr/grammars-v4/tree/master/tinybasic/tinybasic.g4
+grammar_file: https://github.com/antlr/grammars-v4/tree/master/tinybasic/
+tinybasic.g4
 aliases:
```
Comment (jayvdb) [inline data/Language/TinyBasic.yaml]: syntax error: could not find expected ':'

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/JavaScript.yaml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 identifier: JavaScript
 wikidata: Q2005
-grammar_file: https://github.com/antlr/grammars-v4/tree/master/javascript/JavaScriptParser.g4
+grammar_file: https://github.com/antlr/grammars-v4/tree/master/javascript/
+JavaScriptParser.g4
 aliases:
```
Comment (jayvdb) [inline data/Language/JavaScript.yaml]: syntax error: could not find expected ':'

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/Python.yaml [hunk 4]
```diff
@@ -1,7 +1,8 @@
 identifier: Python
 wikidata: Q28865
 versions: 2.7, 3.3, 3.4, 3.5, 3.6
-grammar_file: https://github.com/antlr/grammars-v4/tree/master/python3/Python3.g4
+grammar_file: https://github.com/antlr/grammars-v4/tree/master/python3/
+Python3.g4
 aliases:
```
Comment (jayvdb) [inline data/Language/Python.yaml]: syntax error: could not find expected ':'

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/CSharp.yaml [hunk 5]
```diff
@@ -1,7 +1,8 @@
 identifier: CSharp
 full_name: C#
 wikidata: Q2370
-grammar_file: https://github.com/antlr/grammars-v4/tree/master/csharp/CSharpParser.g4
+grammar_file: https://github.com/antlr/grammars-v4/tree/master/csharp/
+CSharpParser.g4
 aliases:
```
Comment (jayvdb) [inline data/Language/CSharp.yaml]: syntax error: could not find expected ':'

*Origin: YAMLLintBear, Section: `yaml`.*

File: .coafile [hunk 6]
```diff
@@ -0,0 +1,16 @@
+[Default]
```
Comment (jayvdb) [inline .coafile]: an empty Default is useless, and `Default` is deprecated anyway ... just remove

File: .coafile [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+[autopep8]
+bears = PEP8Bear, PycodestyleBear
```
Comment (jayvdb) [inline .coafile]: these are python bears, and there is no python files in this repo

File: .coafile [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+pycodestyle_ignore = E121, E123, E126, E226, E704,
+                     W503, W504
+
+default_actions = PEP8Bear: ApplyPatchAction
+
+[jinja2]
+bears = Jinja2Bear
+files = **.jinja2, **.jj2
```
Comment (jayvdb) [inline .coafile]: there are no .jj2 files in this repo

File: .coafile [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+[yaml]
+bears = YAMLLintBear
+files = *.yml, **/*.yaml
```
Comment (jayvdb) [inline .coafile]: `**.yaml` does the same thing

File: data/Language/TinyBasic.yaml [hunk 10]
```diff
@@ -1,7 +1,8 @@
 identifier: TinyBasic
 full_name: TinyBasic
 wikidata: Q586622
-grammar_file: https://github.com/antlr/grammars-v4/tree/master/tinybasic/tinybasic.g4
+grammar_file: https://github.com/antlr/grammars-v4/tree/master/tinybasic
+              /tinybasic.g4
```
Comment (jayvdb) [inline data/Language/TinyBasic.yaml]: in other cases you leave the `/` on the previous line.  here it is different.  please be consistent.

File: .coafile (hunk 8 above)
Comment (RazvanMitrica) [reply .coafile]: I saw there were jj2 files on other repos and I thought that those kind of files could be included here in the future as well.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coAST/pull/39/commits/923609de6022ea79443938bd26b5ee97a54bb497

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being fastforwarded automatically. Please **DO NOT** push while fastforward is in progress or your changes would be lost permanently :warning:

**Comment by jayvdb** [comment]
Automated fastforward with [GitMate.io](https://gitmate.io) was successful! :tada:

**Pull request merged**